packages/next/client/index.tsx

```diff
@@ -494,7 +494,8 @@ export function renderError(renderErrorProps: RenderErrorProps): Promise<any> {
 }
 
 let reactRoot: any = null
-let shouldHydrate: boolean = typeof ReactDOM.hydrate === 'function'
+// On initial render a hydrate should always happen
+let shouldHydrate: boolean = true
```

Comment:
The check for `ReactDOM.hydrate` there was done for React 15/16 compat, with React 17 being the minimum version it can now be removed